File: python/list_models.py
import os
from google.genai import client
from dotenv import load_dotenv

load_dotenv()
api_key = os.getenv("GEMINI_API_KEY")

if not api_key:
    print("No API Key found.")
else:
    try:
        c = client.Client(api_key=api_key)
        print("Listing models...")
        # The method might be different depending on the SDK version, trying standard ones
        for m in c.models.list(config={'page_size': 10}):
            print(f"- {m.name}")
    except Exception as e:
        print(f"Error: {e}")

File: monitor_dashboard/monitor_dashboard_vite/functions/list_models.js
const { GoogleGenerativeAI } = require('@google/generative-ai');
const dotenv = require('dotenv');
dotenv.config();

async function list() {
  const genAI = new GoogleGenerativeAI(process.env.GOOGLE_API_KEY);
  try {
    const models = await genAI.listModels();
    console.log(JSON.stringify(models, null, 2));
  } catch (e) {
    console.error(e);
  }
}

list();
